analyzers/src/SonarAnalyzer.CSharp/Rules/LoopsAndLinq.cs

```diff
@@ -120,11 +118,14 @@ private static void CheckIfCanBeSimplifiedUsingSelect(ForEachStatementSyntax for
                 c.ReportDiagnosticWhenActive(diagnostic);
             }
 
-            static IEnumerable<IdentifierNameSyntax> GetIdentifiers(ForEachStatementSyntax forEachStatementSyntax) =>
+            static IEnumerable<IdentifierNameSyntax> GetStatementIdentifiers(ForEachStatementSyntax forEachStatementSyntax) =>
                 forEachStatementSyntax.Statement
                                       .DescendantNodes()
                                       .OfType<IdentifierNameSyntax>()
                                       .Where(identifierNameSyntax => identifierNameSyntax.Identifier.ValueText == forEachStatementSyntax.Identifier.ValueText);
+
+            static bool IsNotLeftSideOfAssignment(IdentifierNameSyntax identifierSyntax) =>
+                !(identifierSyntax.Parent.Parent is AssignmentExpressionSyntax assignment && assignment.Left == identifierSyntax.Parent);
```

Comment:
why not pass `memberAccessExpressionSyntax` here? 

`identifierSyntax.Parent.Parent` is quite hard to understand - when reading this function you need to go back to realize that `identifierSyntax.Parent` is a `MemberAccessExpressionSyntax` ...

if you pass `memberAccessExpressionSyntax` instead of `identifierSyntax` you'll only need

`!(memberAccessExpressionSyntax.Parent is AssignmentExpressionSyntax assignment && assignment.Left == memberAccessExpressionSyntax);`

which is easier to understand